config/v1/types_ingress.go

```diff
@@ -82,8 +82,52 @@ type IngressSpec struct {
 	// Note that if there are no RequiredHSTSPolicies, any HSTS Policy annotation on the route is valid.
 	// +optional
 	RequiredHSTSPolicies []RequiredHSTSPolicy `json:"requiredHSTSPolicies,omitempty"`
+
+	// platform holds desired information specific to the underlying
+	// infrastructure provider for the ingress.
+	Platform *IngressPlatformSpec `json:"platform,omitempty"`
+}
+
+type IngressPlatformSpec struct {
```

Comment:
This is a union so needs a union tag, please also add some description of the struct
```suggestion
// +union
type IngressPlatformSpec struct {
```

Reply:
Please make sure to include a description of the expect use of the union in the godoc